packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1228,19 +1246,126 @@ export default abstract class SqlIntegration
-      , d.${dimensionColumn} as dimension_value
-      , MAX('${dimensionColumn}') as dimension_name
+      variation AS variation
+      , ${dimensionColumn} AS dimension_value
+      , MAX(${this.castToString(`'${dimensionColumn}'`)}) AS dimension_name
       , COUNT(*) AS units
     FROM
-      __distinctUnits d
-    ${whereClause}
+      __distinctUnits
+    ${whereClause ?? ""}
     GROUP BY
-      d.variation
-      , d.${dimensionColumn})`;
+      variation
+      , ${dimensionColumn})`;
+  }
+
+  getAutomaticDimensionQuery(params: AutomaticDimensionQueryParams): string {
+    const exposureQuery = this.getExposureQuery(params.exposureQueryId || "");
+
+    const { baseIdType } = getBaseIdTypeAndJoins([[exposureQuery.userIdType]]);
+
+    const startDate = subDays(new Date(), params.lookbackDays);
+    const timestampColumn = "e.timestamp";
+    return format(
+      `-- Suggest Dimensions
+    WITH
+      __rawExperiment AS (
+        ${compileSqlTemplate(exposureQuery.query, {
+          startDate: startDate,
+        })}
+      ),
+      __experimentExposures AS (
+        -- Viewed Experiment
+        SELECT
+          e.${baseIdType} as ${baseIdType}
+          , e.timestamp
+          ${params.dimensions
+            .map((d) => `, e.${d.id} AS dim_${d.id}`)
+            .join("\n")}
+        FROM
+          __rawExperiment e
+        WHERE
+          ${timestampColumn} >= ${this.toTimestamp(startDate)}
+      ),
+      __distinctUnits AS (
+        SELECT
+          ${baseIdType}
+          ${params.dimensions
+            .map(
+              (d) => `
+            , ${this.getDimensionColumn(baseIdType, d)} AS dim_exp_${d.id}`
+            )
+            .join("\n")}
+          , 1 AS variation
+        FROM
+          __experimentExposures e
+        GROUP BY
+          e.${baseIdType}
+      ),
+      -- One row per dimension slice
+      dim_values AS (
+        (SELECT
+          1 AS variation
+          , ${this.castToString("'All'")} AS dimension_value
+          , ${this.castToString("'All'")} AS dimension_name
+          , COUNT(*) AS units
+        FROM
+          __distinctUnits
+          ) UNION ALL
+        ${params.dimensions
+          .map((d) => this.getUnitCountCTE(`dim_exp_${d.id}`))
+          .join("\nUNION ALL\n")}
+      ),
+      total_n AS (
+        SELECT
+          SUM(units) AS N
+        FROM dim_values
+        WHERE dimension_name = 'All'
+      ),
+      dim_values_sorted AS (
+        SELECT
+          dimension_name
+          , dimension_value
+          , units
+          , ROW_NUMBER() OVER (PARTITION BY dimension_name ORDER BY units DESC) as rn
+        FROM
+          dim_values
+        WHERE
+          dimension_name != 'All'
+      )
+      SELECT
+        dim_values_sorted.dimension_name AS dimension_name,
+        dim_values_sorted.dimension_value AS dimension_value,
+        dim_values_sorted.units AS units,
+        dim_values_sorted.units / n.N * 100 AS percent
```

Comment:
I think this will do integer division and end up as `0` in some databases.  We can cast the numerator to a float first or just select `n.N` here instead of calculate percent in typescript